zkvm/src/blockchain/state.rs

```diff
@@ -0,0 +1,78 @@
+use super::block::{Block, BlockHeader, BlockID};
+use super::errors::BCError;
+use crate::{Entry, TxLog, VMError, UTXO};
+use std::collections::{HashSet, VecDeque};
+
+#[derive(Clone)]
+pub struct BCState {
```

Comment:
I wish there was a nice single-word term. Meanwhile, `BlockchainState` looks milder to my eye than “B-C-State”.